include/util/guidance/turn_lanes.hpp

```diff
@@ -69,7 +69,7 @@ class LaneTuple
     bool operator!=(const LaneTuple other) const;
 
     LaneID lanes_in_turn;
-    LaneID first_lane_from_the_right;
+    LaneID first_lane_from_the_right; // is INVALID_LANEID when no lanes present
```

Comment:
Just a thought: the OSRM convention was to use these special-ids. Whats your take on using optionals for these? Would be a slight performance hit, but I guess it would be a good price to pay for getting rid of these where possible.